runtime/parachains/src/paras/mod.rs

```diff
@@ -683,30 +685,11 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig {
 		fn build(&self) {
-			let mut parachains: Vec<_> = self
-				.paras
-				.iter()
-				.filter(|(_, args)| args.parachain)
-				.map(|&(ref id, _)| id)
-				.cloned()
-				.collect();
-
-			parachains.sort();
-			parachains.dedup();
-
-			Parachains::<T>::put(&parachains);
-
+			let mut parachains = ParachainsCache::new();
 			for (id, genesis_args) in &self.paras {
-				let code_hash = genesis_args.validation_code.hash();
-				<Pallet<T>>::increase_code_ref(&code_hash, &genesis_args.validation_code);
```

Comment:
Previously, it didn't check for `validation.code.0.is_empty()`, but I assume this is never the case for the genesis otherwise it's borked?

Reply:
Yeah, good point, I should've mentioned this explicitly in the description.

I did not care that much about this case. Ultimately, using empty wasm code will fail during candidate validation. However, this is not the only way a PVF can be broken: all PVFs with less than 4-8 bytes are broken. Many others greater than that are at least useless. I'd say vast majority of byte combination are not valid wasm files or compressed blobs. Hence there is no explicit check.

However, you question gave me a thought: we are using an empty value as a sentinel value for a code path that won't be reachable from here. So I think this particular case warrants an empty code check.